### PR TITLE
fix: select all checkbox clears when a new page of data loads

### LINKED ISSUE #2328: Bug: Select All doesn't clear when moving to the next page of contributors
### Describe the bug

If I'm looking at the contributors tab of the insights page, and I select all and then move to the next page, it doesn't clear the select all checkbox. To add more contributors to my list, I have to manually click one by one.

image

### Steps to reproduce

1. Go to https://app.opensauced.pizza/pages/BekahHW/01/contributors?range=30&limit=50
2. Select all
3. Got to page 2

https://github.com/open-sauced/app/assets/34313413/e4b4aa7d-e224-457f-b84f-567d6552d728

### Browsers

Chrome

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by aryanas159**
## Description

I have added a state to check whether the checkbox for 'select all' is checked or not and added the required logic to add users across multiple pages, the same issue was with the add contributors to list page as well as the new list creating page. I have fixed those as well.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #2328 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/114330931/086a84e9-9419-4b81-9061-ac628a07ea11

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
FinallyFinishedGIF (2)

### CODE REVIEW

**Comment by OgDev-01** [comment]
Thanks for this fix @aryanas159 🍕. Can you also apply the same fix to the repositories table if you have the bandwidth?

**Comment by aryanas159** [comment]
> Thanks for this fix @aryanas159 🍕. Can you also apply the same fix to the repositories table if you have the bandwidth?

Sure, I can look into it.

**Comment by aryanas159** [comment]
@OgDev-01 I have made the changes for the repository table.

**Comment by OgDev-01** [comment]
> @OgDev-01 I have made the changes for the repository table.

Thanks for tackling this @aryanas159... I'll take a peak on this shortly 👍

**Review by OgDev-01 — APPROVED** [review]
LGTM... Well done 👏 👏  @aryanas159

**Review by jpmcb — COMMENTED** [review]
Looks good! Approved the build preview and went through it: looks good! I'll leave it to others for code / style review. But LGTM

File: components/organisms/Contributors/contributors.tsx [hunk 1]
```diff
@@ -42,14 +42,21 @@ const Contributors = ({ repositories }: ContributorProps): JSX.Element => {
...
+  const [isSelectAll, setIsSelectAll] = useState(false); // state to check if all contributors are selected
   const contributors = data.map((pr) => {
     return {
       host_login: pr.author_login,
       first_commit_time: pr.updated_at,
     };
   });
-
+  // check if all contributors are selected or not
+  useEffect(() => {
```
Comment (nickytonline) [inline components/organisms/Contributors/contributors.tsx:53]: Any reason this requires a `useEffect`? This should be driven by the check/uncheck in the `onClick` even if the paged data changes.

**Review by nickytonline — CHANGES_REQUESTED** [review]
Any reason this requires a `useEffect`? This should be driven by the check/uncheck in the `onClick` even if the paged data changes.

Instead, when the new page of data loads, the event that changes the page should call `setIsSelectAll(false)`.

**Comment by aryanas159** [comment]
@nickytonline If we call ```setIsSelectAll(false)``` every time the page changes, those elements will also be removed if we move to a page where we have selected all previously. Also the ```useEffect``` ensures that if a user manually selects all the users on a page, then the ```select all``` checkbox will also be checked.

**Comment by nickytonline** [comment]
> @nickytonline If we call `setIsSelectAll(false)` every time the page changes, those elements will also be removed if we move to a page where we have selected all previously. Also the `useEffect` ensures that if a user manually selects all the users on a page, then the `select all` checkbox will also be checked.

This is true; however, that is some select all functionality that all our grids are missing. Currently, if you select all, it's for the current page only, but when we do that, we probably want to give an option like in GMail to select all the data.

CleanShot 2024-01-06 at 15 30 18

**Comment by brandonroberts** [comment]
I agree with @nickytonline. I think we need some more thought on this and should close this for now.

**Comment by nickytonline** [comment]
Thanks for working on this @aryanas159. Once we get the true  select all in place, we can revisit this.

**Pull request closed without merging**